### PR TITLE
cellprofiler 3.1.9

### PR DESCRIPTION
**Pull request by sunyi000**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/cellprofiler/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - python <3
+    - boto3
+      #    - centrosome==1.1.6
+      #    - docutils==0.14
+    - h5py
+      #    - inflect==2.1.0
+    - wxpython
+    - joblib
+    - mahotas
+    - matplotlib==2.2.3
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: I don't think this strict pin is necessary.

File: recipes/cellprofiler/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    - mysqlclient
+    - numpy
+    - javabridge
+    - prokaryote
+    - python-bioformats
+    - pyzmq=15.3.0
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: relax the pin?

File: recipes/cellprofiler/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    - raven
+    - requests
+    - scikit-image=0.14.0
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: `- scikit-image 0.14.*`

File: recipes/cellprofiler/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    - scikit-learn
+    - scipy
+      #    - six==1.11.0
+    - pillow
+    - networkx==2.2
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: `- networkx 2.2.*`

File: recipes/cellprofiler/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    - vigra
+    - pytest
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: Is pytest really a runtime dependency?

File: recipes/cellprofiler/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+      #    - dask==0.19.1
+
+test:
+  imports:
+    - cellprofiler
+
+about:
+  summary: "CellProfiler is free, open-source software"
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: Please add license, doc_url etc and under extras you as a maintainer.

File: recipes/cellprofiler/meta.yaml (hunk 1 above)
Comment (sunyi000) [reply recipes/cellprofiler/meta.yaml]: yes, it needs to be > 2.0.0, so I will change it to >=2.0

File: recipes/cellprofiler/meta.yaml (hunk 2 above)
Comment (sunyi000) [reply recipes/cellprofiler/meta.yaml]: removed the pin

File: recipes/cellprofiler/meta.yaml (hunk 5 above)
Comment (sunyi000) [reply recipes/cellprofiler/meta.yaml]: i moved it to `host:` as I can see setup.py requires it

File: recipes/cellprofiler/meta.yaml (hunk 6 above)
Comment (sunyi000) [reply recipes/cellprofiler/meta.yaml]: added. I have push the actual license file with the correct content yet. I will do that soon

**Comment by sunyi000** [comment]
I have made another commit and it builds fine locally.

but I still used the javabridge package from the old 'free' channel.

I will have a look at #7701 see what I can do.

Yi

**Comment by sunyi000** [comment]
Hi @bgruening ,

a couple of questions,

I tried to install javabridge to local channel using its own conda recipe mentioned in #7710, then build and install cellprofiler. It seems ok, so is this the correct approach for now, since javabridge is not yet in conda-forge yet?

is that possible to write the javabridge installation from source within cellprofiler's meta.yaml?
for exmaple, create a build.sh for cellprofiler ,download javabridge source, and do python setup.py install? 

thanks very much,
Yi

File: recipes/cellprofiler/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,80 @@
+{% set name = "CellProfiler" %}
+{% set version = "3.1.9" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+    git_url: "https://github.com/CellProfiler/CellProfiler.git"
```
Comment (bgruening) [inline recipes/cellprofiler/meta.yaml]: here you should use a tarball

**Comment by drpatelh** [comment]
Hi @bgruening ! Sorry, I completely missed your ping and just saw this again 😏 The javabridge issue seemed insurmountable at the time so I ended up using Docker containers directly provided by CellProfiler in the end. Hope you and loved ones are well.

**Comment by bgruening** [comment]
All good @drpatelh! 

Conda: https://anaconda.org/bioconda/cellprofiler
Container: https://quay.io/repository/biocontainers/cellprofiler?tag=latest&tab=tags
Galaxy Integration: https://usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/cp_cellprofiler/cp_cellprofiler/3.1.9

Thanks to @sunyi000 and @beatrizserrano!

**Pull request closed without merging**